libreflect/src/stack_setup.c

```diff
@@ -26,7 +31,8 @@ void synthetic_auxv(size_t *auxv)
 	auxv[12] = AT_SECURE;
     // Required for stack cookies on glibc, hope your payload doesn't get popped
 	auxv[14] = AT_RANDOM; auxv[15] = (size_t)auxv;
-	auxv[16] = AT_NULL; auxv[17] = 0;
+	auxv[16] = AT_SYSINFO_EHDR; auxv[17] = at_sysinfo_ehdr_value;
```

Comment:
I'm not super familiar with this code; But it looks like the convention is to assign the actual value inside of `load_program_info`, unless i'm missing something 👀

Reply:
load_program_info()` loads data from the parsed ELFs `exe` and `interp` and puts it in the auxilliary vector. The value for `AT_SYSINFO_EHDR` does not come from any parsed ELF at all. It simply comes from the running process itself and the original auxiliary vector that process (so `noexec`) started with. Similarly `AT_RANDOM` is also set directly and not in `load_program_info()`.

So `load_program_info()` is for putting values into the auxilliary vector that come from the parsed ELFs.